src/sentry/api/event_search.py

```diff
@@ -1221,6 +1231,16 @@ def visit_has_filter(
         # the key is has here, which we don't need
         negation, _, _, _, (search_key,) = children
 
+        # Some datasets do not support the !has filter, but we allow
+        # team_key_transaction because we control that field and special
+        # case the way it's processed in search
+        if (
+            not self.config.allow_not_has_filter
+            and is_negated(negation)
+            and search_key.name != TEAM_KEY_TRANSACTION_ALIAS
```

Comment:
Special case `TEAM_KEY_TRANSACTION_ALIAS` because we control that property with a field converter